### PR TITLE
Add modify_notes custom action

### PR DESCRIPTION
**Pull request by lderequesensS**
- [x] unit test passed
- [x] less than x (800 lines?)

## Purpose

Add new custom action to modify notes on an alert

[(Jira ticket link)](https://zf.atlassian.net/browse/ZFE-94526)

## Approach

- Add new action to zerofox_connector.py
- Change metadata on zerofox.json

## Where should the reviewer start?

zerofox_connector.py

## This PR addresses a:

- [ ] Bug
- [x] Feature
- [ ] Technical Debt
- [ ] Chore
- [ ] Vulnerability

## Quality assurance (mandatory)

Explain what you did to ensure the right behavior of the changes

## Steps to validate/test  
  
1.  Go to https://phantom.zf-integrations.com/
2. On the "Open" widget you can see alerts
3. Click one and go to "Analyst" (upper right corner)
4. Click the artifact that is there and copy the "Source ID"
5. Click "Action" (a little bit lower than Analyst)
6. On action choose "modify notes" and choose zerofox3 asset
7. Put the alert ID, choose between append or replace and write some text
8. Click "Launch"

### CODE REVIEW

File: zerofox_connector.py [hunk 1]
```diff
@@ -832,6 +835,88 @@ def _threat_submit(self, param):
...
+            return action_result.set_status(phantom.APP_ERROR)
+
+        action = param.get("modify_action", "append")
+        previous_notes = alert.get("notes", "")
+        notes = param.get("notes", "")
+        new_notes = ""
+        if action == "replace":
+            new_notes = notes
+        elif action == "append":
+            new_notes = previous_notes + notes
```
Comment (DNRRomero) [inline zerofox_connector.py:880]: faltaria un salto de linea o algo para separarlos, 
si las notas antiguas son
"some notes"
y las otras son 
"some other notes"

deberian quedar en lineas separadas

**Review by DNRRomero — CHANGES_REQUESTED** [review]
se ve bien! duda, esto pasa la pipeline de splunk soar?

File: zerofox_connector.py (hunk 1 above)
Comment (lderequesensS) [reply zerofox_connector.py:880]: Te refieres a dejar esto como:
```
new_notes += previous_notes
new_notes += notes
```
u otra cosa?

Comment (DNRRomero) [reply zerofox_connector.py:880]: me refiero a 
`new_notes = notes if not previous_notes else f"{previous_notes}\n{notes}"`

Comment (DNRRomero) [reply zerofox_connector.py:880]: la suma solo concatena strings, pero no considera el salto de linea

**Review by DNRRomero — APPROVED** [review]

**Review by zkto — APPROVED** [review]

**Comment by lderequesensS** [comment]
Cerrado porque ya existe un PR en Splunk con este código.

**Pull request closed without merging**